src/virtual_dom/listener.rs

```diff
@@ -1,11 +1,56 @@
 use super::Ev;
 use crate::app::MessageMapper;
 use crate::browser::{dom::lifecycle_hooks::fmt_hook_fn, util::ClosureNew};
-use enclose::enclose;
-use std::{fmt, mem};
+use std::{fmt, mem, rc::Rc};
 use wasm_bindgen::{closure::Closure, JsCast};
 
-type EventHandler<Ms> = Box<dyn FnMut(web_sys::Event) -> Ms>;
+pub struct EventHandler<Ms>(Rc<dyn Fn(web_sys::Event) -> Ms>);
+
+impl<Ms, F: Fn(web_sys::Event) -> Ms + 'static> From<F> for EventHandler<Ms> {
+    fn from(func: F) -> Self {
+        EventHandler(Rc::new(func))
+    }
+}
+
+impl<Ms> Clone for EventHandler<Ms> {
+    fn clone(&self) -> Self {
+        EventHandler(self.0.clone())
+    }
+}
+
+impl<Ms> PartialEq for EventHandler<Ms> {
+    fn eq(&self, other: &EventHandler<Ms>) -> bool {
+        Rc::ptr_eq(&self.0, &other.0)
+    }
+}
+
+// TODO: try to print Ms too, maybe by restricting Ms: Debug ?
+impl<Ms> fmt::Debug for EventHandler<Ms> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("EventHandler<_>")
+    }
+}
```

Comment:
Is `fmt::Debug` impl for `EventHandler` necessary? And we shouldn't restrict `Ms` to `Debug`. It should be possible to pass `EventHandler` into `log!` if you explicitly restrict `Ms: Debug` in your application (i.e. in the root) if you are on `nightly` Rust - see https://github.com/seed-rs/seed/pull/234/files

Reply:
> Is fmt::Debug impl for EventHandler necessary?

Good point, some why I thought this was needed... we can remove this if it's not needed.

Reply:
I think that `log!` support for implicit `Debug` objects on nightly is enough. 
Remove it, if you want or I'll probably remove it during VDOM refactor.